scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -0,0 +1,544 @@
+-  Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
+- Masks variance within a sample to highlight variance across samples.
+
+We are staring with the simple comparison of looking at one cell type and comparing across two groups of samples, samples belonging to either the ERMS and ARMS subtypes. 
+When we do this, we will first want to pseudo-bulk our dataset to group our cells by cell type and by sample.
+This will create a new `SingleCellExperiment` object that contains the pseudo-bulked counts to be used as input for differential expression analysis. 
+Then we can subset the pseudo-bulked object to just contain our cell type of interest before proceeding. 
+
+Before we apply pseudo-bulking to our dataset, lets look at a simple example of how pseudo-bulking works. 
+
+```{r}
+# create an example counts matrix
+counts_mtx <- matrix(1:12, 
+                     ncol = 4,
+                     dimnames = list(c("geneA", "geneB", "geneC"),
+                                     c("A-cell1", "A-cell2", "B-cell1", "B-cell2")))
+counts_mtx
+```
+
+
+```{r}
+# create new matrix, summing the counts for each gene across all cells in the specified group
+pseudobulked_counts <- cbind(
+  rowSums(counts_mtx[,1:2]), # sum all counts for cells labeled with group A
+  rowSums(counts_mtx[,3:4]) # sum all counts for cells labeled with group B
+)
+pseudobulked_counts 
+```
+Now the actual pseudo-bulking for our dataset! 
+We can pseudo-bulk using any groups that we are interested in.
+For right now, we would are interested in looking at gene expression across cell types, so we want to group the counts matrix by both cell type and sample. 
+
+```{r}
+# first let's subset our coldata to only have the columns we care about in pseudobulking 
+pseudobulked_coldata <- colData(rms_sce)[, c("celltype_broad", "sample")]
+
+# this creates a new SCE object that contains the pseudobulked counts across the provided groups 
+pseudobulked_sce <- scater::aggregateAcrossCells(rms_sce, 
+                                                  id = pseudobulked_coldata)
+
+# column names aren't automatically added, so let's add them in 
+colnames(pseudobulked_sce) <- paste(pseudobulked_sce$celltype_broad, 
+                                    pseudobulked_sce$sample, sep = "-")
+
+pseudobulked_sce
+```
+
+How does the new pseudo-bulked `SingleCellExperiment` look different? 
+
+```{r}
+# only 37 columns in the counts assay 
+counts(pseudobulked_sce)[1:10, 1:10]
+```
+
+Let's take a look at what the `colData` now looks like in the pseudo-bulked SCE object. 
+
+```{r}
+# note the new addition of the column with number of cells per group 
+colData(pseudobulked_sce)
+```
+
+### Filtering the pseudo-bulked `SingleCellExperiment` object
+
+*Add a bit more justification on filtering and why we need it*
+
+```{r}
+# filter to only the HVG list
+hv_genes <- metadata(integrated_sce)$combined_hvg
```

Comment:
Yea I wasn't entirely sure about this one... Looking at the OSCA chapter they show the DE analysis example on an object that has been merged and corrected and only contains the HVGs. But the way they choose HVGs is slightly different, instead of using `getTopHVGs` they retain anything with a positive biological component of the variance. So they retain a lot more than 2000 genes. 
Here's the link to how they set up the data that they used for reference: http://bioconductor.org/books/3.16/OSCA.multisample/chimeric-mouse-embryo-10x-genomics.html#chimeric-mouse-embryo-10x-genomics

I think filtering to only the 2000 genes may be too stringent, but I think we definitely want to do some sort of filtering.

Reply:
I might try it without filtering... I think the main reason for filtering is to speed calculations, but I don't expect that to be too bad.